rare/components/tabs/games/__init__.py

```diff
@@ -159,146 +105,90 @@ def __init__(self, parent=None):
-        self.filter_games("")
-
-    def installation_started(self, app_name: str):
-        self.installing_widget.set_game(app_name)
-
-        i_widget, l_widget = self.widgets.get(app_name, (None, None))
-        if not i_widget or not l_widget:
-            return
-        i_widget.setVisible(False)
-        l_widget.setVisible(False)
-
-    def verification_finished(self, igame: InstalledGame):
-        # only if igame needs verification
-        i_widget, l_widget = self.widgets[igame.app_name]
-        i_widget.igame = igame
-        l_widget.igame = igame
-
-        i_widget.leaveEvent(None)
-        l_widget.update_text()
+    @pyqtSlot()
+    def scroll_to_top(self):
+        self.icon_view_scroll.verticalScrollBar().setSliderPosition(
+            self.icon_view_scroll.verticalScrollBar().minimum()
+        )
+        self.list_view_scroll.verticalScrollBar().setSliderPosition(
+            self.list_view_scroll.verticalScrollBar().minimum()
+        )
 
     @pyqtSlot()
     def show_import(self):
-        self.setCurrentWidget(self.integrations_tabs)
-        self.integrations_tabs.show_import()
+        self.setCurrentWidget(self.integrations_page)
+        self.integrations_page.show_import()
 
     @pyqtSlot()
     def show_egl_sync(self):
-        self.setCurrentWidget(self.integrations_tabs)
-        self.integrations_tabs.show_egl_sync()
+        self.setCurrentWidget(self.integrations_page)
+        self.integrations_page.show_egl_sync()
 
     @pyqtSlot()
     def show_eos_ubisoft(self):
-        self.setCurrentWidget(self.integrations_tabs)
-        self.integrations_tabs.show_eos_ubisoft()
+        self.setCurrentWidget(self.integrations_page)
+        self.integrations_page.show_eos_ubisoft()
 
-    def show_game_info(self, app_name):
-        self.game_info_tabs.update_game(app_name)
-        self.setCurrentWidget(self.game_info_tabs)
-
-    def show_uninstalled_info(self, game):
-        self.uninstalled_info_tabs.update_game(game)
-        self.setCurrentWidget(self.uninstalled_info_tabs)
+    @pyqtSlot(RareGame)
+    def show_game_info(self, rgame):
+        self.setCurrentWidget(self.game_info_page)
+        self.game_info_page.update_game(rgame)
 
     @pyqtSlot()
     def update_count_games_label(self):
-        self.head_bar.set_games_count(len(self.core.get_installed_list()), len(self.game_list))
-
-    def setup_game_list(self):
-        self.update_count_games_label()
+        self.head_bar.set_games_count(
+            len(self.core.get_installed_list()), len(self.api_results.games + self.api_results.na_games)
+        )
 
-        # add installed games
-        for igame in sorted(self.core.get_installed_list(), key=lambda x: x.title):
-            icon_widget, list_widget = self.add_installed_widget(igame.app_name)
-            self.icon_view.layout().addWidget(icon_widget)
-            self.list_view.layout().addWidget(list_widget)
+    # FIXME: Remove this when RareCore is in place
```

Comment:
It is not as far as `RareGame` instantiation is concerned. `RareCore` is basically a dumb holder right now rather the backend library it will be when it is merged.